docs/guides/initializers.md

```diff
@@ -0,0 +1,123 @@
+---
+jupytext:
+  formats: ipynb,md:myst
+  text_representation:
+    extension: .md
+    format_name: myst
+    format_version: 0.13
+    jupytext_version: 1.13.8
+kernelspec:
+  display_name: Python 3
+  name: python3
+---
+
++++ {"id": "5D-VCn-KtpXk"}
+
+# Initializers
+
+In this guide, we will demonstrate how to use initializers and some of the common pitfalls when using initializers in Flax.
+
++++ {"id": "6nsUREar3k1H"}
+
+`Initializers` are functions that can be passed as optional arguments to the kernel initializer (`kernel_init`) and the bias initializer (`bias_init`) if you want to specify how the parameters of a Module layer are initialized. A full list of Flax initializers can be found [here](https://flax.readthedocs.io/en/latest/api_reference/flax.linen.html#module-flax.linen.initializers), and are in fact, the same as the [JAX initializers](https://jax.readthedocs.io/en/latest/jax.nn.initializers.html).
+
+The default kernel initializer is [`flax.linen.initializers.lecun_normal`](https://flax.readthedocs.io/en/latest/api_reference/_autosummary/flax.linen.initializers.lecun_normal.html) and the default bias initializer is [`flax.linen.initializers.zeros`](https://flax.readthedocs.io/en/latest/api_reference/_autosummary/flax.linen.initializers.zeros.html).
+
+```{code-cell}
+---
+executionInfo:
+  elapsed: 118
+  status: ok
+  timestamp: 1673133040623
+  user:
+    displayName: Marcus Chiam
+    userId: '17531616275590396120'
+  user_tz: 480
+id: 8Gvqoi_rzII5
+---
+import jax, jax.numpy as jnp
+import flax.linen as nn
+from flax.linen.initializers import lecun_normal, zeros
+```
+
+```{code-cell}
+---
+executionInfo:
+  elapsed: 61
+  status: ok
+  timestamp: 1673133040851
+  user:
+    displayName: Marcus Chiam
+    userId: '17531616275590396120'
+  user_tz: 480
+id: nHyw_azR0TAL
+---
+layer = nn.Dense(features=5)
+default_params = layer.init(jax.random.PRNGKey(42), jnp.empty((1,4)))['params']
+
+layer = nn.Dense(features=5, kernel_init=lecun_normal(), bias_init=zeros)
+manual_params = layer.init(jax.random.PRNGKey(42), jnp.empty((1,4)))['params']
+
+# should get the same param values
+assert (default_params['kernel'] == manual_params['kernel']).all()
+assert (default_params['bias'] == manual_params['bias']).all()
```

Comment:
Currently all this shows is what the defaults of the init functions are, which one can also just see by looking at the code, so maybe it is not worth adding a code snippet for it.

Maybe it is more interesting to demonstrate how Flax uses the initial rng to generate the params? Also I would always use `np.testing.assert_allclose` when comparing JAX arrays.

```suggestion
x = jnp.empty((1,4))
rng = random.PRNGKey(0)
params = nn.Dense(features=5).init(rng, x)['params']

# Internally, Flax splits off a new RNG from the original one for each param
# using a counter which starts at 1.
kernel_rng = scope.LazyRng.create(rng, 1).as_jax_rng()

np.testing.assert_allclose(params['kernel'], lecun_normal()(kernel_rng, (4, 5)))
np.testing.assert_allclose(params['bias'], jnp.empty((5,)))
```